frontend/packages/schema-editor/src/components/SchemaInspector/ItemDataComponent.tsx

```diff
@@ -144,6 +146,9 @@ export function ItemDataComponent(props: IItemDataComponentProps) {
   const handleArrayPropertyToggle = () => mutate(toggleArrayField(data, pointer));
 
   const handleChangeNodeName = () => {
+     if (!isTextModified) {
```

Comment:
You also need to reset this state so that it is ready for the next update from the user.
You can avoid using state entirely here by instead just moving the check on l.156-157 up to the top of this function and flipping it:
```suggestion
     const staleName = getNameFromPointer({ pointer });
    if (staleName === nodeName) {
```

Then you can remove the check from l156-157 😊

Reply:
done